mirascope/core/base/_create.py

```diff
@@ -196,10 +206,15 @@ def inner(
                 *args: _P.args, **kwargs: _P.kwargs
             ) -> TCallResponse | _ParsedOutputT:
                 fn_args = get_fn_args(fn, args, kwargs)
-                dynamic_config = get_dynamic_configuration(fn, args, kwargs)
+                if (
+                    dynamic_config := get_dynamic_configuration(fn, args, kwargs)
+                ) and "client" in dynamic_config:
+                    dynamic_client = dynamic_config.get("client")
+                else:
```

Comment:
ditto here on the else

Reply:
I fixed it

Reply:
i updated this to use `nonlocal` and just set `client` so it's a little cleaner / easier to read